server/etcdserver/server.go

```diff
@@ -1592,14 +1592,13 @@ func (s *EtcdServer) mayRemoveMember(id types.ID) error {
 	}
 
 	// protect quorum if some members are down
-	m := s.cluster.VotingMembers()
-	active := numConnectedSince(s.r.transport, time.Now().Add(-HealthInterval), s.MemberID(), m)
-	if (active - 1) < 1+((len(m)-1)/2) {
+	since := time.Now().Add(-HealthInterval)
+	if !isConnectedToQuorumSince(s.r.transport, since, s.MemberID(), s.cluster.Members()) {
 		lg.Warn(
 			"rejecting member remove request; local member has not been connected to all peers, reconfigure breaks active quorum",
 			zap.String("local-member-id", s.MemberID().String()),
 			zap.String("requested-member-remove", id.String()),
-			zap.Int("active-peers", active),
+			zap.Int("active-peers", numConnectedSince(s.r.transport, since, s.MemberID(), s.cluster.Members())),
```

Comment:
not a fan of this calculation because it repeats what is done inside `isConnectedToQuorumSince` and the number of connected member may be different at the time of the second calculation.

Reply:
I see, thanks.